jujugui/static/gui/src/app/components/icon-list/icon-list.js

```diff
@@ -0,0 +1,105 @@
+/* Copyright (C) 2018 Canonical Ltd. */
+'use strict';
+
+const PropTypes = require('prop-types');
+const React = require('react');
+
+/**
+  Display a list of icons for an entity.
+*/
+class IconList extends React.Component {
+  constructor() {
+    super();
+  }
+
+  /**
+    Generate the store state for the item.
+    @param {String} id The entity id.
+    @returns {Object} The store state.
+  */
+  _generateStoreState(id) {
+    // TODO frankban: it should be clear whether this id is legacy or not.
+    let url;
+    try {
+      url = window.jujulib.URL.fromLegacyString(id);
+    } catch(_) {
+      url = window.jujulib.URL.fromString(id);
+    }
+    return {
+      profile: null,
+      search: null,
+      store: url.path()
+    };
+  }
+
+  /**
+    Generate the store URL for an entity.
+    @param {String} id The entity id.
```

Comment:
It'd be nice if internally we only pass URL objects, rather than strings. Strings are always ambiguous (is it a new style URL, and old one, does it have the schema, etc).

Reply:
Yeah this would be nice....I'm not sure how that would work in practice though as we frequently have just dealt with strings which are returned from the api.